test/lib/ansible_test/_internal/commands/sanity/__init__.py

```diff
@@ -123,6 +123,21 @@
 SANITY_ROOT = os.path.join(ANSIBLE_TEST_CONTROLLER_ROOT, 'sanity')
 TARGET_SANITY_ROOT = os.path.join(ANSIBLE_TEST_TARGET_ROOT, 'sanity')
 
+# This should use documentable plugins from constants instead
+UNDOCUMENTED_PLUGIN_TYPES = set([
+    # not supported by ansible-doc
+    'action',
+    'doc_fragments',
+    'filter',
+    'module_utils',
+    'terminal',
+    'test',
+    # The following are plugin directories not directly supported by ansible-core (and thus also not by ansible-doc)
+    # (https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst#modules--plugins)
+    'plugin_utils',
+    'sub_plugins',
+])
```

Comment:
I would not create new constant but reuse the DOCUMENTABLE_PLUGINS or CONFIGURABLE_PLUGINS ones from ansible itself.

Reply:
Ref: https://github.com/ansible/ansible/pull/71734#discussion_r487525585 from over a year ago :)

I'll try again to figure out whether it works better now.

> FYI, action plugins are documented via modules of the same name, see 'template'.

That's why action plugins must be listed here, since they themselves do not contain documentation.

Reply:
When trying to import this when running the sanity tests inside the default Docker container, I get:
```
[...]
  File "/root/ansible/test/lib/ansible_test/_internal/commands/sanity/ansible_doc.py", line 9, in <module>
    from ansible.constants import DOCUMENTABLE_PLUGINS
ModuleNotFoundError: No module named 'ansible'
```
While the workaround suggested by @mattclay in https://github.com/ansible/ansible/pull/71734#discussion_r490513071 would work for validate_modules, it doesn't work for the ansible-doc sanity test (if I understood it correctly).

If this is the only problem with this PR, my suggestion would be to keep the list (because this PR just moves it from one place to another in ansible-test, and starts using that list in another place).

Reply:
my issue is that the lists will fall out of sync and one is additive while the other is subtractive ... fun all around in future.

Reply:
if direct import does not work, maybe a query to ansible-doc to get 'supported' types is in order?

Reply:
I did rename the list to `DOCUMENTABLE_PLUGINS` and added a note to ansible.constants.DOCUMENTABLE_PLUGINS that also this list needs to be updated. That way it should be easier to keep them in sync.